src/common/utils/buildUrl.ts

```diff
@@ -35,3 +35,24 @@ export function buildURLPositional<P extends object = {}, Q extends object = {}>
     return buildURL(path, { params, query, fragment });
   };
 }
+
+export type UrlParamsFor<Pathname extends string> =
+  Pathname extends `${string}/:${infer A}?/${infer Tail}`
+    ? Partial<Record<A, string>> & UrlParamsFor<`/${Tail}`>
+    : Pathname extends `${string}/:${infer A}/${infer Tail}`
+      ? Record<A, string> & UrlParamsFor<`/${Tail}`>
+      : Pathname extends `${string}/:${infer A}?`
+        ? Partial<Record<A, string>>
+        : Pathname extends `${string}/:${infer A}`
+          ? Record<A, string>
+          : {};
```

Comment:
Ew. Wasn't there a lib somewhere which externalised icky stuff like this? :)

Reply:
I think I remember that there was, but I don't know if it is worth it for just this type.

Reply:
Sure. Maybe let's not go and duplicate something like this in the future :) Resolved.